app/assets/javascripts/initializers/initializeAllFollowButts.js

```diff
@@ -1,104 +1,58 @@
-function initializeAllFollowButts() {
-  var followButts = document.getElementsByClassName('follow-action-button');
-  for (var i = 0; i < followButts.length; i++) {
-    initializeFollowButt(followButts[i]);
-  }
-}
-
-//private
+'use strict';
 
-function initializeFollowButt(butt) {
-  var user = userData();
-  var deviceWidth = window.innerWidth > 0 ? window.innerWidth : screen.width;
-  var buttInfo = JSON.parse(butt.dataset.info);
-  var userStatus = document
-    .getElementsByTagName('body')[0]
-    .getAttribute('data-user-status');
-  if (userStatus === 'logged-out') {
-    addModalEventListener(butt);
-    return;
-  }
-  if (buttInfo.className === 'Tag' && user) {
-    handleTagButtAssignment(user, butt, buttInfo);
-    return;
+function addFollowText(butt, style) {
+  var btnToAddFollowText = butt;
+  if (style === 'small') {
+    btnToAddFollowText.textContent = '+';
+  } else if (style === 'follow-back') {
+    btnToAddFollowText.textContent = '+ FOLLOW BACK';
   } else {
-    if (butt.dataset.fetched === 'fetched') {
-      return;
-    }
-    fetchButt(butt, buttInfo);
+    btnToAddFollowText.textContent = '+ FOLLOW';
   }
 }
 
-function addModalEventListener(butt) {
-  assignState(butt, 'login');
-  butt.onclick = function(e) {
-    e.preventDefault();
-    showModal('follow-button');
-    return;
-  };
-}
-
-function fetchButt(butt, buttInfo) {
-  butt.dataset.fetched = 'fetched';
-  var dataRequester;
-  if (window.XMLHttpRequest) {
-    dataRequester = new XMLHttpRequest();
+function addFollowingText(butt, style) {
+  var btnToAddFollowingText = butt;
+  if (style === 'small') {
+    btnToAddFollowingText.textContent = '✓';
   } else {
-    dataRequester = new ActiveXObject('Microsoft.XMLHTTP');
+    btnToAddFollowingText.textContent = '✓ FOLLOWING';
   }
-  dataRequester.onreadystatechange = function() {
-    if (
-      dataRequester.readyState === XMLHttpRequest.DONE &&
-      dataRequester.status === 200
-    ) {
-      addButtClickHandle(dataRequester.response, butt);
-    }
-  };
-  dataRequester.open(
-    'GET',
-    '/follows/' + buttInfo.id + '?followable_type=' + buttInfo.className,
-    true,
-  );
-  dataRequester.send();
 }
 
-function addButtClickHandle(response, butt) {
-  // currently lacking error handling
-  var buttInfo = JSON.parse(butt.dataset.info);
-  assignInitialButtResponse(response, butt);
-  butt.onclick = function(e) {
-    e.preventDefault();
-    handleOptimisticButtRender(butt);
-  };
-}
-
-function handleTagButtAssignment(user, butt, buttInfo) {
-  var buttAssignmentBoolean =
-    JSON.parse(user.followed_tags)
-      .map(function(a) {
-        return a.id;
-      })
-      .indexOf(buttInfo.id) !== -1;
-  var buttAssignmentBoolText = buttAssignmentBoolean ? 'true' : 'false';
-  addButtClickHandle(buttAssignmentBoolText, butt);
-  shouldNotFetch = true;
```

Comment:
This variable didn't seem to be defined anywhere throughout the repo, so removed it.